### DISCUSSION TITLE
Why the input of int64 automatically converted to int32 when the model is loaded?

### DISCUSSION THREAD

**Discussion by LinGeLin** [openvinotoolkit/openvino#24067] [Q&A] [answered]
I work on an extension. The input must be int64. But when I debug it, I find that the input will be converted to int32. Is there any special mechanism to control it? I have observed that the validate_and_infer_types function is called many times. May I ask where in ov will this function of the extension plugin be called at runtime? And from the log I printed, the first few times the input is still int64, but later it becomes int32, which is very confusing.

![image](https://github.com/openvinotoolkit/openvino/assets/24861783/af3636b3-601c-4900-8063-7dc9b1bdbbc8)

version 2024.0.0

**Comment by ilya-lavrenov** [comment]
Please, create an issue instead of discussion.

**Reply by LinGeLin** [reply, marked as answer]
Is this a bug?

**Reply by LinGeLin** [reply]
created. 
https://github.com/openvinotoolkit/openvino/issues/24072

**Comment by LinGeLin** [comment]
![image](https://github.com/openvinotoolkit/openvino/assets/24861783/66cfd1de-ab68-465c-b555-3edf2858708e)


int64 input is not actually supported? Will it convert to int32?

**Reply by ilya-lavrenov** [reply]
are you running on GPU?
Screen from GPU code.

**Reply by LinGeLin** [reply]
sorry, I misread it, running on the cpu

**Reply by LinGeLin** [reply]
![image](https://github.com/openvinotoolkit/openvino/assets/24861783/7399e62c-6230-4483-82fc-4a1a2b4e19b4)

The CPU code also has the same setting, which will convert i64 to i32

**Reply by LinGeLin** [reply]
Can I remove this line of code and recompile openvino to ensure that int64 is not converted to int32?

**Reply by ilya-lavrenov** [reply]
yes. You can either change CPU plugin code or assume that i32 is coming to your operation.

Currently, OpenVINO CPU plugin changes the precisions for better performance, but it does not assume that precisions for custom operations should be preserved. It's a bug and I've assigned people from CPU team to have a look at it

**Reply by LinGeLin** [reply]
https://github.com/openvinotoolkit/openvino/blob/34825fe2c2ccda54f8b6c2afac3a724e730a526b/src/common/transformations/include/transformations/convert_precision.hpp#L43
```
 * List of operations that are supported by this transformations for i64 -> i32 conversion:
 *     opset4::Parameter
 *     opset4::Convert
 *     opset4::ShapeOf
 *     opset4::Range
 *     opset3::NonMaxSuppression
 *     opset4::NonMaxSuppression
 *     opset4::TopK
 *     opset4::NonZero
 *     opset4::Bucketize
```

In my usage scenario, all INT64s before the custom operator cannot be converted. But I've noticed that some op are supported by the i64 - > i32 conversion.  Is it that even if I remove the code which setting conver precision, if I use these operators, the input to int64 will be converted to int32 ?

**Reply by ilya-lavrenov** [reply]
CC @nshchego

**Reply by LinGeLin** [reply]
My program is now running properly and I have commented out the i64->i32 code in transformation_pipeline.cpp. The next step is to verify the performance of the existing production model. thank u
